code/modules/surgery/organs/blood.dm

```diff
@@ -95,9 +95,11 @@
 		..()
 		if(dna.species.exotic_blood)
 			var/datum/reagent/R = GLOB.chemical_reagents_list[get_blood_id()]
-			if(istype(R) && isturf(loc))
-				R.reaction_turf(get_turf(src), amt * EXOTIC_BLEED_MULTIPLIER)
-
+			if(EXOTIC_COLOR in dna.species.species_traits)
+				if(istype(R) && isturf(loc))
+					R.reaction_turf(get_turf(src), amt * EXOTIC_BLEED_MULTIPLIER, dna.species.blood_color)
+				else
+					R.reaction_turf(get_turf(src), amt * EXOTIC_BLEED_MULTIPLIER)
```

Comment:
Why this one here? Both this line and 100 call `reaction_turf`. This one can also runtime since `R` does not have to be a reagent here